### PR TITLE
Create `actor_categories` schema and API

### PR DESCRIPTION
**Pull request by parndt**
Implements #24  

We need a many to many join table describing a relationship between
`actors`, introduced for #18, and `categories`, from impact.

##### Permissions by role

- public: none
- registered, no role: none
- analyst: R
- manager: CRD
- admin: CRD

##### Rules

- Only for actors where `category`'s taxonomy is enabled for `actortype`

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -8,6 +8,7 @@
   end
   get "static_pages/home"
 
+  resources :actor_categories, except: [:edit, :update]
```
Comment (tmfrnz) [inline config/routes.rb:11]: @parndt curious about the `except` bit here that appears to be inconsistent with other, similar routes (eg `measure_categories`)

File: config/routes.rb [hunk 2]
```diff
@@ -8,6 +8,7 @@
...
   resources :actor_measures, only: [:index, :show]
```
Comment (tmfrnz) [inline config/routes.rb:12]: @parndt unrelated, but also I am wondering about the role of `only` here for `actor_measures` (as well as on `measure_actors`) as these should also allow creating and deleting via the API. Also I notice that the `create` and `destroy` operations are also missing in the controllers: 
- https://github.com/dumparkltd/marine-defrag-server/blob/master/app/controllers/measure_actors_controller.rb
- https://github.com/dumparkltd/marine-defrag-server/blob/master/app/controllers/actor_measures_controller.rb

Or maybe I am missing something?

**Review by tmfrnz — APPROVED** [review]

File: config/routes.rb (hunk 1 above)
Comment (parndt) [reply config/routes.rb:11]: `resources` defines 7 routes: `index`, `show`, `edit`, `update`, `new`, `create`, `destroy` so here we are saying we want them all `except: `[:edit, :update]`

From the [Rails routing guide](https://guides.rubyonrails.org/routing.html#crud-verbs-and-actions):
image

This is the same as saying

```suggestion
  resources :actor_categories, only: [:index, :show, :new, :create, :destroy]
```

Happy to switch it to `only` as it matches the others

Comment (tmfrnz) [reply config/routes.rb]: thanks for the details and update! happy either way =)

File: config/routes.rb (hunk 2 above)
Comment (tmfrnz) [reply config/routes.rb:12]: resolving...

see #20, #21

**Pull request merged**